openslides/poll/models.py

```diff
@@ -119,10 +122,19 @@ def print_votescast(self):
         return value
 
     def get_percent_base(self):
-        if self.get_percent_base_choice() == "WITHOUT_INVALID" and self.votesvalid and self.votesvalid > 0:
+        if self.get_percent_base_choice() == "WITHOUT_INVALID" and\
+                self.votesvalid and self.votesvalid > 0:
             base = 100 / float(self.votesvalid)
-        elif self.get_percent_base_choice() == "WITH_INVALID" and self.votescast and self.votescast > 0:
+        elif self.get_percent_base_choice() == "WITH_INVALID" and\
+                self.votescast and self.votescast > 0:
             base = 100 / float(self.votescast)
+        # elif self.get_percent_base_choice() == "WITHOUT_ABSTAIN" and\
+        #     self.votesvalid and self.votesvalid > 0:
+        #     try:
+        #         base = 100 / float(self.votesvalid - abstain)
+        #         #TODO get abstains from server
```

Comment:
The `CollectDefaultVotesMixin` is a mixin that should work for different poll/vote class systems in the respective apps. In this case you can not access the abstain votes in this abstract layer because there is not guarantee that the concrete poll class (MotionPoll e. g.) has abstention votes at all. What you have to do is to override the method `get_percent_base` in both the `MotionPoll` class and the `AssignmentPoll` class. Then you can substract the abstention easily in motion and you have top decide what should happen in the different assignment cases.